File: dialogloadingLibrary/src/main/java/com/tobey/dialogloading/GifDialog.java
package com.tobey.dialogloading;

import android.app.Activity;
import android.app.Dialog;
import android.content.Context;
import android.content.res.Resources;
import android.graphics.BitmapFactory;
import android.graphics.drawable.Drawable;
import android.text.TextUtils;
import android.util.Log;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.core.content.ContextCompat;

import java.lang.ref.SoftReference;

import pl.droidsonroids.gif.GifImageView;

/**
 * @Package: com.tobey.dialogloading
 * @ClassName: GifDialog
 * @Author: Tobey_r1
 * @CreateDate: 2021/4/13 14:22
 * @Description: java类作用描述
 * @UpdateUser: 更新者
 * @UpdateDate: 2021/4/13 14:22
 * @UpdateRemark: 更新说明
 * @Version: 1.0
 */
public class GifDialog {

    private static String TAG = "com.tobey.gitdialog";

    public static boolean isclose = false;
    private static TextView tipTextView;
    private static GifImageView img;
    private static LinearLayout gifLayout;
    private static Context mContext;

    public static Dialog createDialog(Context context, String msg) {
        isclose = false;
        mContext = context;
        LayoutInflater inflater = LayoutInflater.from(context);
        View v = inflater.inflate(R.layout.dialog_gif_loading, null);// 得到加载view
        LinearLayout layout = v.findViewById(R.id.loading_view);// 加载布局
        tipTextView = v.findViewById(R.id.loading_message);// 提示文字
        gifLayout = v.findViewById(R.id.gifLayout);
        tipTextView.setText(msg);// 设置加载信息
        img = v.findViewById(R.id.giftView);
        Dialog loadingDialog = new Dialog(context, R.style.MyDialogStyle);// 创建自定义样式dialog
        loadingDialog.setCancelable(true); // 是否可以按“返回键”消失
        loadingDialog.setCanceledOnTouchOutside(false); // 点击加载框以外的区域
        loadingDialog.setContentView(layout, new LinearLayout.LayoutParams(
                LinearLayout.LayoutParams.MATCH_PARENT,
                LinearLayout.LayoutParams.MATCH_PARENT));
        // 设置布局
        /**
         *将显示Dialog的方法封装在这里面
         */
        Window window = loadingDialog.getWindow();
        WindowManager.LayoutParams lp = window.getAttributes();
        lp.width = WindowManager.LayoutParams.MATCH_PARENT;
        lp.height = WindowManager.LayoutParams.MATCH_PARENT;
        //lp.type = lp.TYPE_PHONE;
        window.setGravity(Gravity.CENTER);
        window.setAttributes(lp);
        window.setWindowAnimations(R.style.PopWindowAnimStyle);
        loadingDialog.show();
        return loadingDialog;
    }

    public static void closeDialog(Dialog mDialogUtils) {

        if (mDialogUtils != null && mDialogUtils.isShowing()) {
            mDialogUtils.dismiss();
            isclose = true;
        }
    }


    /** 设置显示消息颜色
     * @param color 文本颜色
     */
    public static void setTextColor(int color){
        tipTextView.setTextColor(mContext.getResources().getColor(color));
    }

    /** 设置弹窗窗体颜色
     * @param color 文本颜色
     */
    public static void setlayoutBackGroundColor(int color){
        gifLayout.setBackgroundColor(color);
    }

    /** 设置弹窗圆角颜色
     * @param drawable 图片资源
     */
    public static void setlayoutBackGround(Drawable drawable){
        gifLayout.setBackground(drawable);
    }

    /** 设置显示消息颜色
     */
    public static void setImageBackGround(int stlyle){
        if (isDrawable(stlyle,mContext)){
            img.setImageResource(stlyle);
        }

    }

    private static boolean isDrawable(int id,Context context){
        BitmapFactory.Options options = new BitmapFactory.Options();
        Resources res = context.getResources();
        SoftReference<BitmapFactory.Options> softReference = new SoftReference<>(options);
        if (softReference !=null){
            softReference.get().inJustDecodeBounds = true;
            BitmapFactory.decodeResource(res,id,options);
            String type = options.outMimeType;
            if (TextUtils.isEmpty(type)) {
                Log.e(TAG,"未能识别的图片");
                return false;
            } else if (type.substring(6, type.length()).equals("gif")){
                return true;
            }else {
                type = type.substring(6, type.length());
                Log.e("image type -> ", type);
                return false;
            }

        }else {
            Log.e(TAG,"内存不足，自动回收");
            return false;
        }

    }


}

File: dialogloadingLibrary/src/main/java/com/tobey/dialogloading/GifDialog.kt
package com.tobey.dialogloading

import android.annotation.SuppressLint
import android.app.Dialog
import android.content.Context
import android.graphics.BitmapFactory
import android.graphics.drawable.Drawable
import android.util.Log
import android.view.Gravity
import android.view.LayoutInflater
import android.view.WindowManager
import android.widget.LinearLayout
import android.widget.TextView
import pl.droidsonroids.gif.GifImageView

@SuppressLint("StaticFieldLeak")
object GifDialog {

       private var tipTextView: TextView? = null
       private  var mContext: Context? = null
       private var img: GifImageView? = null
       private var gifLayout: LinearLayout? = null

       fun createDialog(context: Context?, msg: String?): Dialog {
              mContext = context
              val inflater = LayoutInflater.from(context)
              val v = inflater.inflate(R.layout.dialog_gif_loading, null) // 得到加载view
              val layout = v.findViewById<LinearLayout>(R.id.loading_view) // 加载布局
              tipTextView = v.findViewById<TextView>(R.id.loading_message)
              tipTextView!!.text = msg
              gifLayout = v.findViewById(R.id.gifLayout)
              img = v.findViewById(R.id.giftView)
              val loadingDialog = Dialog(context!!, R.style.MyDialogStyle) // 创建自定义样式dialog
              loadingDialog.apply {
                     setCancelable(true)
                     setCanceledOnTouchOutside(false)
                     setContentView(
                            layout, LinearLayout.LayoutParams(
                                   LinearLayout.LayoutParams.MATCH_PARENT,
                                   LinearLayout.LayoutParams.MATCH_PARENT
                            )
                     )
              }.show()
              /**
              * 将显示Dialog的方法封装在这里面
              */
              val window = loadingDialog.window
              val lp = window!!.attributes
              lp.width = WindowManager.LayoutParams.MATCH_PARENT
              lp.height = WindowManager.LayoutParams.MATCH_PARENT
              window.apply {
                     setGravity(Gravity.CENTER)
                     attributes = lp
                     setWindowAnimations(R.style.PopWindowAnimStyle)
              }
              return loadingDialog
       }

       fun closeDialog(mDialogUtils: Dialog?) {
              if (mDialogUtils != null && mDialogUtils.isShowing) {
                     mDialogUtils.dismiss()
              }
              img?.clearFocus()
              img = null
              tipTextView = null
              gifLayout = null
              mContext = null
       }

       /** 设置显示消息颜色
        * @param color 文本颜色
        */
       fun setTextColor(color: Int) {
              tipTextView!!.setTextColor(mContext!!.resources.getColor(color))
       }

       /** 设置弹窗窗体颜色
        * @param color 文本颜色
        */
       fun setlayoutBackGroundColor(color: Int) {
              gifLayout!!.setBackgroundColor(color)
       }

       /** 设置弹窗圆角颜色
        * @param drawable 图片资源
        */
       fun setlayoutBackGround(drawable: Drawable?) {
              gifLayout!!.background = drawable
       }

       /** 设置显示消息颜色
        */
       fun setImageBackGround(stlyle: Int) {
              if (isDrawable(stlyle, mContext)) {
                     img!!.setImageResource(stlyle)
              }
       }

       private fun isDrawable(id: Int, context: Context?): Boolean {
              val options = BitmapFactory.Options()
              val res = context!!.resources
              options.inJustDecodeBounds = true
              BitmapFactory.decodeResource(res, id,options)
              val type = options.outMimeType
              if (type.isNotEmpty()){
                     return  when(type.substring(6, type.length)){
                            "gif" ->  true
                            else -> {
                                   Log.e("image type -> ", type)
                                   false
                            }
                     }
              }
              Log.e(TAG, "内存不足，自动回收")
              return false
       }
}

